BackupPlugin.inc.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/**
+ * @file plugins/generic/backup/BackupPlugin.inc.php
+ *
+ * Copyright (c) 2014-2018 Simon Fraser University
+ * Copyright (c) 2000-2018 John Willinsky
+ * Distributed under the GNU GPL v2. For full terms see the file docs/COPYING.
+ *
+ * @class BackupPlugin
+ * @ingroup plugins_generic_backup
+ *
+ * @brief Plugin to allow generation of a backup extract
+ */
+
+import('lib.pkp.classes.plugins.GenericPlugin');
+
+class BackupPlugin extends GenericPlugin {
+	/**
+	 * @copydoc Plugin::register()
+	 */
+	public function register($category, $path, $mainContextId = null) {
+		if (parent::register($category, $path, $mainContextId)) {
+			$this->addLocaleData();
+			return true;
+		}
+		return false;
+	}
+
+	/**
+	 * Get the display name of this plugin
+	 * @return string
+	 */
+	public function getDisplayName() {
+		return __('plugins.generic.backup.name');
+	}
+
+	/**
+	 * Get the description of this plugin
+	 * @return string
+	 */
+	public function getDescription() {
+		return __('plugins.generic.backup.description');
+	}
+
+	/**
+	 * Override the builtin to get the correct template path.
+	 *
+	 * @return string Plugin template path
+	 */
+	public function getTemplatePath($inCore = false) {
+		return parent::getTemplatePath($inCore) . 'templates/';
+	}
+
+	/**
+	 * Designate this plugin as a site plugin
+	 */
+	public function isSitePlugin() {
+		return true;
+	}
+
+	/**
+	 * Get plugin JS URL
+	 *
+	 * @return string Public plugin JS URL
+	 */
+	public function getJsUrl($request) {
+		return $request->getBaseUrl() . '/' . $this->getPluginPath() . '/js';
+	}
+
+	/**
+	 * @see Plugin::getActions()
```

Comment:
Use `@copydoc` unless you intend to re-describe the parameters and return value.